app/pages/patientdata/patientdata.js

```diff
@@ -317,26 +356,13 @@ export let PatientData = React.createClass({
           );
 
       case 'settings':
+        return this.renderSettings();
 
-        return (
-          <Settings
-            bgPrefs={this.state.bgPrefs}
-            chartPrefs={this.state.chartPrefs}
-            currentPatientInViewId={this.props.currentPatientInViewId}
-            timePrefs={this.state.timePrefs}
-            patientData={this.state.processedPatientData}
-            onClickRefresh={this.handleClickRefresh}
-            onClickNoDataRefresh={this.handleClickNoDataRefresh}
-            onSwitchToBasics={this.handleSwitchToBasics}
-            onSwitchToDaily={this.handleSwitchToDaily}
-            onSwitchToModal={this.handleSwitchToModal}
-            onSwitchToSettings={this.handleSwitchToSettings}
-            onSwitchToWeekly={this.handleSwitchToWeekly}
-            trackMetric={this.props.trackMetric}
-            uploadUrl={this.props.uploadUrl}
-            ref="tideline" />
-          );
-
+      case 'settingsprintview':
+        // set here so that we default back settings and not print
+        this.setState({ chartType: 'settings' });
```

Comment:
So this is the `setState()`-in-`render()` pattern that React is particularly not a fan of. (since `render()` is called immediately following any `setState()`, this can often lead to a cyclical call and kill everything). Since we call the same `renderSettings()` after this regardless, is there any harm in moving the `window.print()` call directly to the `handleSwitchToSettingsPrintView` function (replacing the `setState()` there) and just dropping this whole `case`? [I tested it quickly, and it seemed to function as I expected and format the printout identically, but I haven't dug real deep into things I might have missed]